test/tests.py

```diff
@@ -1624,12 +1631,14 @@ def analyze_c_files(self, files, func_spans, cblock_spans, not_linemarked):
                         # Need to reverse as .pop() pops from the end
                         curr_cblock_expected.reverse()
                         continue
-                    assert (curr_func in expected_func_lines
-                            and dml_lineno in expected_func_lines[curr_func]),\
-                           (f'In {curr_file}: Unexpected DML line '
-                            + f'{dml_lineno} at {c_lineno} '
-                            + (f'(inside {curr_func})' if curr_func else
-                               '(top-level)'))
+                    if (curr_func not in expected_func_lines
+                        or dml_lineno not in expected_func_lines[curr_func]):
+                        raise self.Error([
+                            (Path(self.filename).resolve(), dml_lineno,
+                             "Unexpected DML line"),
```

Comment:
I think this would be easier to parse by switching the order of the messages and talk about the C part first:
`f"Line ({'inside ' + curr_func if curr_func else 'at top level'}) redirected to unexpected DML line"`
and then
`f"Unexpected DML line"`